internal/services/fetch.go

```diff
@@ -2,27 +2,100 @@ package services
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
-	"github.com/sirupsen/logrus"
+	"io"
 	"net/http"
 	"strconv"
+	"strings"
 	"time"
+
+	"github.com/be-ys/Genesis/internal/structures"
+	"github.com/sirupsen/logrus"
 )
 
-func fetch(service string, url string) string {
+func fetch(service string, url string, headers map[string]string, extract map[string]structures.ExtractData, format string) string {
 	start := time.Now()
-	client := http.Client{Timeout: 5 * time.Second}
-	res, err := client.Get(url)
-	var responseCode = res.StatusCode
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		logrus.Warningf("Error trying to request %s: %s", url, err.Error())
+	}
+
+	// Add headers to http request
+	if len(headers) != 0 {
```

Comment:
I think we don't need to check length as we are using a for each loop just below :smile: